### DISCUSSION TITLE
Solution TCP implementation always sends 512 bytes payload

### DISCUSSION THREAD

**Discussion by kmc7468** [ANLAB-KAIST/KENSv3#223] [Project Q&A 2. Transport Layer (TCP)] [answered]
As I know, in `TestEnv_Any.TestTransfer_Connect_Recv_Symmetric` test case, my implementation is the client and the solution implementation is the server.

When I set the receive buffer size at least 512 bytes, there is no error. However, when I set it below 512 bytes, the test cases related to receiving fail.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/18364980/6add97ee-13dd-4a50-9366-67675bb8f9ab)

I noticed that the solution always send 512 bytes payload even though my implementation advertised that its receive window size is below 512 bytes.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/18364980/d1160626-64ec-4063-9e92-ddbc7c942d22)

Is there any mistake in the solution? Or Do I have to accept only 511 bytes (, which is the window size) from 512 bytes payload?

**Comment by Pusnow** [comment, marked as answer]
Yes, it is probably a bug in the TCP solution implementation.
I'll contact the original author of the implementation to discuss whether it is intended behavior.

Until then, please use buffer larger than 512. 

Thank you for reporting this issue.
